apps/confessions-server/src/routing/routes/confessionRoutes.ts

```diff
@@ -0,0 +1,152 @@
+import express, { Request, Response, NextFunction } from "express";
+import { sign } from "jsonwebtoken";
+import {
+  SemaphoreGroupPCDPackage,
+  SerializedSemaphoreGroup
+} from "@pcd/semaphore-group-pcd";
+import { Group } from "@semaphore-protocol/group";
+import { sha256 } from "js-sha256";
+import { ApplicationContext } from "../../types";
+import { SEMAPHORE_GROUP_URL, ACCESS_TOKEN_SECRET, authenticateJWT } from "../../util/auth";
+import { prisma } from "../../util/prisma";
+
+export function initConfessionRoutes(
+  app: express.Application,
+  _context: ApplicationContext
+): void {
+  // this endpoint verifies the proof is valid,
+  // and the proof claim matches the semaphoreGroupUrl and the confession in the request
+  // before saving the confession to the database.
+  app.post("/confessions", async (req: Request, res: Response, next: NextFunction) => {
```

Comment:
a better name for this route could be `"/new-confession"`

Reply:
Another thing is that technically this route could just use the `jwt` to authenticate, right? Having both authentication mechanisms in this example app may be confusing, but I'll leave it to you to decide whether it makes sense.

Reply:
I think another issue of verifying this proof for posting a confession is that if the proof was generated a while ago, then the proof may be outdated, since new ppl have been added to the group since then. that's not ideal.

Reply:
Changed the route name.

Technically this route could use the `jwt` to authenticate. Still want to keep it to use pcd for demonstration purposes, since this is a demo app for developers. The pcd will verify the user belongs to the zuzalu group and also made the confession (a signal).
I split the routes file to `authedRoutes` which contains `/login` and get `/confessions`. It shows how authenticate with jwt would work. And `pcdRoutes` which contains this `/new-confession`. It shows how to pcd can be used.

Reply:
> I think another issue of verifying this proof for posting a confession is that if the proof was generated a while ago, then the proof may be outdated, since new ppl have been added to the group since then. that's not ideal.

This is definitely an issue.
However, it is a general issue which applies to /login too, and any pcd using the semaphore group proof in general.
We could snapshot the group merkletree whenever there's changes to the group. However, if it happens frequently, it could be problematic.
How do we solve the problem for the zuzalu passport? When a new zuzalu is added, do we update the content fetched from the semaphore_group_url? @ichub